build_tools/cmake/rebuild.sh

```diff
@@ -47,19 +47,20 @@ CMAKE_ARGS=(
   # get a reasonable mix of {builds with asserts, builds with other features
   # such as ASan but without asserts}.
   "-DIREE_ENABLE_ASSERTIONS=ON"
-
-  # Enable CUDA compiler and runtime builds unconditionally. Our CI images all
-  # have enough deps to at least build CUDA support and compile CUDA binaries
-  # (but not necessarily test on real hardware).
-  "-DIREE_HAL_DRIVER_CUDA=ON"
-  "-DIREE_TARGET_BACKEND_CUDA=ON"
 )
 
+if [[ -z "${IREE_CUDA_DISABLE_BUILD}" ]]; then
+  CMAKE_ARGS+=(
+    # Enable CUDA compiler and runtime builds unless disabled. Our CI images all
+    # have enough deps to at least build CUDA support and compile CUDA binaries.
+    "-DIREE_HAL_DRIVER_CUDA=ON"
+    "-DIREE_TARGET_BACKEND_CUDA=ON"
+  )
+fi
+
 "$CMAKE_BIN" "${CMAKE_ARGS[@]?}" "$@" ..
-echo "Building all"
-echo "------------"
-"$CMAKE_BIN" --build . -- -k 0
 
-echo "Building test deps"
-echo "------------------"
-"$CMAKE_BIN" --build . --target iree-test-deps -- -k 0
+TARGETS="all iree-test-deps"
+echo "Building targets: ${TARGETS}"
+echo "-------------------------------------------"
+"$CMAKE_BIN" --build . --target ${TARGETS} -- -k 0
```

Comment:
OTOH, we lose parallelism between the builds. Any evidence on how big that effect is?

Reply:
OK, sure. Just to check: I don't fully understand what `-k 0` does but it is in principle telling the build to "keep going" in case of errors. Is that consistent with the design here?

Reply:
It's "keep going" for N failures, but with no long name, unfortunately. 0 is a special sentinel value

> -k N     keep going until N jobs fail (0 means infinity) [default=1]